becquerel/core/fitting.py

```diff
@@ -0,0 +1,806 @@
+        # max_idx = np.argmax(counts)
+        # counts[counts < counts[max_idx] * 0.5] *= 0
+        # amp = np.sum(counts) / 0.8
+        # mu = x[max_idx]
+        # sigma = np.sqrt(np.sum(counts * (x - mu)**2) / amp) * 2
+        # if x is None:
+        #     x = np.arange(0, len(y))
+
+        xspan = x[-1] - x[0]
+        mu = x[0] + xspan * center_ratio
+        msk = ((x >= (mu - xspan * width_ratio)) &
+               (x <= mu + xspan * width_ratio))
+        # NOTE: this integration assumes y is NOT normalized to dx (NOW IT IS)
+        amp = np.sum(y[msk]/dx[msk])
+        # c = y[msk]
+        # amp = np.sum(c) - (c[0] + c[-1]) * np.sum(msk) * 0.5
+        # TODO: update this, minimizer creates NaN's if default sigma used (0)
+        sigma = xspan * width_ratio / 10.
+        return [
+            ('{}amp'.format(self.prefix), 'value', amp),
+            ('{}amp'.format(self.prefix), 'min', 0.0),
+            ('{}mu'.format(self.prefix), 'value', mu),
+            ('{}mu'.format(self.prefix), 'min', x[0]),
+            ('{}mu'.format(self.prefix), 'max', x[-1]),
+            ('{}sigma'.format(self.prefix), 'value', sigma),
+            ('{}sigma'.format(self.prefix), 'min', 0.0),
+        ]
+
+
+class ErfModel(Model):
+
+    def __init__(self, *args, **kwargs):
+        super(ErfModel, self).__init__(erf, *args, **kwargs)
+
+    def guess(self, y, x=None, dx=None, center_ratio=0):
+        xspan = x[-1] - x[0]
+        mu = x[0] + xspan * center_ratio
+        return [
+            ('{}amp'.format(self.prefix), 'value', y[0] - y[-1]),
+            ('{}mu'.format(self.prefix), 'value', mu),
+            ('{}sigma'.format(self.prefix), 'expr', 'gauss_sigma'),
+        ]
+
+
+class GaussErfModel(Model):
+
+    def __init__(self, *args, **kwargs):
+        super(GaussErfModel, self).__init__(gausserf, *args, **kwargs)
+        self.set_param_hint(
+            '{}fwhm'.format(self.prefix),
+            expr='{} * {}sigma'.format(FWHM_SIG_RATIO, self.prefix))
+
+    def guess(self, y, x=None, dx=None, center_ratio=0.5, width_ratio=0.5,
+              amp_ratio=0.9):
+        assert center_ratio < 1, \
+            'Center mask ratio cannot exceed 1: {}'.format(center_ratio)
+        assert width_ratio < 1, \
+            'Width mask ratio cannot exceed 1: {}'.format(width_ratio)
+        if x is None:
+            x = np.arange(0, len(y))
+        if dx is None:
+            dx = np.ones_like(x)
+        xspan = x[-1] - x[0]
+        mu = x[0] + xspan * center_ratio
+        msk = ((x >= (mu - xspan * width_ratio)) &
+               (x <= mu + xspan * width_ratio))
+        amp = np.sum(y[msk]/dx[msk])
+        sigma = xspan * width_ratio / 10.
+        return [
+            ('{}ampgauss'.format(self.prefix), 'value', amp*amp_ratio),
+            ('{}ampgauss'.format(self.prefix), 'min', 0.0),
+            ('{}amperf'.format(self.prefix), 'value', amp*(1.0 - amp_ratio)),
+            ('{}amperf'.format(self.prefix), 'min', 0.0),
```

Comment:
The `erf` function should allow for a negative amplitude.

Reply:
You can change the value but this default comes from most cases where you want the `amp>=0`.